src/viewer.js

```diff
@@ -546,89 +539,97 @@ class VolumeImageViewer {
     }
 
     // Metadata Tiles types checks for each instance
-    // look for channels
-    for (let i = 0; i < this[_options].metadata.length; ++i) {
-      const instanceMetadata = formatMetadata(this[_options].metadata[i])
-      if (instanceMetadata.SamplesPerPixel !== 1) {
-        // this is not a monochorme channel, but a color image.
-        continue
-      }
-      if (instanceMetadata.DimensionOrganizationType === '3D' || instanceMetadata.DimensionOrganizationType === '3D_TEMPORAL') {
-        // 3D data
-        // TO DO: get some example data.
-        throw new Error('Volume Image Viewer does hot hanlde 3D channel data yet.')
-      } else if (instanceMetadata.DimensionOrganizationType === 'TILED_FULL') {
-        if (instanceMetadata.TotalPixelMatrixFocalPlanes !== 1) {
-          continue
-        } else {
-          const pathIdentifier = instanceMetadata.OpticalPathSequence[0].OpticalPathIdentifier
-          const channel = this[_channels].find(channel => {
-            return channel.blendingInformation.opticalPathIdentifier === pathIdentifier
-          })
-          if (channel) {
-            channel.addMetadata(this[_options].metadata[i])
+    // group channels by OpticalPathIdentifier
+    const channelsMicroscopyImages = groupMonochormeMetadataInstances(this[_options].metadata)
+    // do additional checks and create channel objects
+    for (let i = 0; i < channelsMicroscopyImages.length; ++i) {
+      const channelMicroscopyImages = channelsMicroscopyImages[i]
+      for (let j = 0; j < channelMicroscopyImages.length; ++j) {
+        const channelMicroscopyImage = channelMicroscopyImages[j]
+        if (channelMicroscopyImage.DimensionOrganizationType === '3D' || channelMicroscopyImage.DimensionOrganizationType === '3D_TEMPORAL') {
+          // 3D data
+          // TO DO: get some example data.
```

Comment:
Let's track such TODOs in issues rather than comments in the source code.

Reply:
we already have it https://github.com/MGHComputationalPathology/slim/issues/11, I will just remove the TO DO line